### DISCUSSION TITLE
whisper 音频文件转文字特别慢， 是否支持Java ？

### DISCUSSION THREAD

**Discussion by jhx1024** [openai/whisper#1430] [Q&A]
两个疑问：
1.使用whisper  xxx.mp3(或其他格式音频文件)特别慢。一首3分钟歌差不多要二三十分钟才能结束。

2.是否支持Java ？

**Comment by phineas-pta** [comment]
please use english

**Comment by Tex2002ans** [comment]
I translated @jhx1024 's post using DeepL:

Chinese->English:

- - -

> **whisper audio files to text is particularly slow, is Java supported? #1430**
>
> Two questions:
>
> 1. Using whisper xxx.mp3 (or other format audio file) is particularly slow. It takes about 20-30 minutes to finish a 3 minute song.
>
> 2. Does it support Java?

- - -

1] Whisper, on the CPU, is quite slow. It is mostly designed to go very fast using a powerful GPU.

If you want to run it faster, there is lots of discussion in this thread:

- https://github.com/openai/whisper/discussions/716

where people have created other projects, like:

- `faster-whisper`

which run with less RAM and can achieve similar results in a fraction of the time.

If you can only run CPU, you may also be limited to some of the smaller models:

- `tiny`
- `base`
- `small`
- `medium`

Trying to run `large-v2` will be MUCH, MUCH, slower.

2] No. Not that I'm aware of. Whisper normally runs in:

- Python

and there are a few other projects that take it and wrap it in C++ or Java (for Android/iOS ports). See some of the discussion here:

- https://github.com/openai/whisper/discussions?discussions_q=is%3Aopen+android
- https://github.com/openai/whisper/discussions?discussions_q=is%3Aopen+ios

**Comment by jhx1024** [comment]
Thank you for answering my question.
